tests/adapter/dbt/tests/adapter/incremental/test_incremental_merge_exclude_columns.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+from collections import namedtuple
+
+
+models__merge_exclude_columns_sql = """
+{{ config(
+    materialized = 'incremental',
+    unique_key = 'id',
+    incremental_strategy='merge',
+    merge_exclude_columns=['msg']
+) }}
+
+{% if not is_incremental() %}
```

Comment:
Does the test case need to be made this way? I feel like it'd make more sense to split this into two SQL statements. By all means, offer a counterperspective. Incremental hacks aren't my strong suit so this may seem very straightforward to use/other users 🙇‍♀️

Reply:
I'm not sure i know of another way! Since this is testing whether or not an incremental model behaves the way we would expect under certain conditions, I would assume we'd need to have the incremental logic scoped to one single model rather than separate sql statements. 

The other incremental adapter tests we have ([here, for one](https://github.com/dbt-labs/dbt-core/blob/e8da84fb9e177d9eee3d7722d3d6906bb283183d/tests/adapter/dbt/tests/adapter/incremental/test_incremental_predicates.py)) use a similar approach

Reply:
@VersusFacit Putting two versions of the data into a single SQL statement is pretty common in our test cases from what I've seen.

Here's one way to peek at a sample:
```shell
git grep "is_incremental" tests/
```

What is less common here is the ordering of the SQL versus the actual execution order. Most of the time, the SQL in our incremental tests appears in this order:
1. 2nd version (incremental)
1. 1st version (full refresh)

What's nice here is that the SQL order matches the execution order:
1. 1st version (full refresh)
1. 2nd version (incremental)

Reply:
Yep, rethought the nature of incremental models and entirely agree.